src/reaction.rs

```diff
@@ -563,6 +576,43 @@ Here's my footer -- bob@example.net"
         Ok(())
     }
 
+    async fn expect_incoming_reactions_event(
+        t: &TestContext,
+        expected_msg_id: MsgId,
+        expected_contact_id: ContactId,
+        expected_reaction: &str,
+    ) -> Result<()> {
+        let event = t
+            .evtracker
+            .get_matching(|evt| matches!(evt, EventType::IncomingReaction { .. }))
+            .await;
+        match event {
+            EventType::IncomingReaction {
+                msg_id,
+                contact_id,
+                reaction,
+            } => {
+                assert_eq!(msg_id, expected_msg_id);
+                assert_eq!(contact_id, expected_contact_id);
+                assert_eq!(reaction, Reaction::from(expected_reaction));
+            }
+            _ => unreachable!(),
+        }
+        Ok(())
+    }
+
+    fn has_incoming_reactions_event(t: &TestContext) -> bool {
```

Comment:
Looks like an acceptable ad-hoc solution, but it just returns `false` on error and is limited to testing reaction events. I'd suggest to add instead smth like this to `test_utils.rs:impl EventTracker`:
```Rust
    /// Consumes emitted events returning the first matching one if any.                                                                                                                                             
    pub async fn get_matching_opt<F: Fn(&EventType) -> bool>(&self, ctx: &Context, event_matcher: F) -> Option<EventType> {
        ctx.emit_event(EventType::Test);                                                                                                                                                                             
        loop {
            let event = self.recv().await.unwrap();
            if event_matcher(&event.typ) {
                return Some(event.typ);
            }
            if let EventType::Test = event.typ {
                return None;
            }
        }
    }
```
(didn't test, only compiled)

Reply:
thanks, i added that snippet, seems still to work :)